general/chainerrl/baselines/action_wrappers.py

```diff
@@ -0,0 +1,340 @@
+    '''
+    Generate an order of queries.
+    '''
+    if prioritized_elements is not None:
+        priorities = copy.copy(prioritized_elements)
+        for key in original_order:
+            if key in prioritized_elements:
+                continue
+            priorities.append(key)
+    else:
+        priorities = copy.copy(original_order)
+    return priorities
+
+
+def generate_query(env_name, priorities, reverse_keys, discrete_indices,
+                   max_camera_range, num_camera_discretize, allow_pitch=False):
+    '''
+    Generate queries.
+    '''
+    query = []
+    for key in priorities:
+        if key == 'camera':
+            # yaw
+            assert num_camera_discretize % 2 == 1
+            half_n = num_camera_discretize // 2
+            scale = max_camera_range * 2 / (num_camera_discretize - 1)
+            for i in range(half_n):
+                query.append(('less', -max_camera_range + scale * (0.5 + i),
+                              'camera_yaw',
+                              discrete_indices[key] + i))
+                query.append(('more', max_camera_range - scale * (0.5 + i),
+                              'camera_yaw',
+                              discrete_indices[key] + num_camera_discretize - 2 - i))
+            if allow_pitch:
+                # pitch
+                for i in range(half_n):
+                    query.append(('less', -max_camera_range + scale * (0.5 + i),
+                                  'camera_pitch',
+                                  discrete_indices[key] + num_camera_discretize - 1 + i))
+                    query.append(('more', max_camera_range - scale * (0.5 + i),
+                                  'camera_pitch',
+                                  discrete_indices[key] + num_camera_discretize * 2 - 3 - i))
+        elif key in NUM_ENUM_ACTIONS[env_name]:
+            n = NUM_ENUM_ACTIONS[env_name][key]
+            for i in range(1, n):
+                query.append(('same', i,
+                              key, discrete_indices[key] + i - 1))
+        elif key in BINARY_KEYS:
+            # BINARY_KEYS
+            if key in reverse_keys:
+                query.append(('less', 0.5,
+                              key, discrete_indices[key]))
+            else:
+                query.append(('more', 0.5,
+                              key, discrete_indices[key]))
+        else:
+            raise ValueError('Unknown key {}.'.format(key))
+    return query
+
+
+def generate_discrete_converter(env_name,
+                                prioritized_elements=None,
+                                always_keys=None,
+                                reverse_keys=None,
+                                exclude_keys=None,
+                                exclude_noop=False,
+                                allow_pitch=False,
+                                max_camera_range=10,
+                                num_camera_discretize=3):
```

Comment:
maybe 7 should be the default since it's what you use

Reply:
This is because this function is also used by DDDQN etc.. Their discretization corresponds to the `num_camera_discretize=3` case. We modified it to `7` on scripts.

Reply:
I see, no problem